### PR TITLE
Code clean up and refactoring

### PR DESCRIPTION
**Pull request by btensay**
Code change is made to do code clean up and refactoring. The code change includes renaming some unit tests. 

https://tools.hmcts.net/jira/browse/ECM-748

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by gauravchawla73463 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/claimsbyhearingvenue/ClaimsByHearingVenueReport.java [hunk 1]
```diff
@@ -1,45 +1,34 @@
...
-        var submitEvents = reportDataSource.getData(
-            UtilHelper.getListingCaseTypeId(searchParams.getCaseTypeId()),
-                searchParams.getDateFrom(), searchParams.getDateTo());
-        var claimsByHearingVenueReportData = initReport(searchParams.getCaseTypeId());
-
-        setReportListingDate(claimsByHearingVenueReportData, searchParams.getDateFrom(),
-                searchParams.getDateTo(), hearingDateType);
-
-        claimsByHearingVenueReportData.setReportPrintedOnDescription(getReportedOnDetail(userFullName));
+    public ClaimsByHearingVenueReportData generateReport(ClaimsByHearingVenueReportDataSource dataSource,
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/claimsbyhearingvenue/ClaimsByHearingVenueReport.java:21]: dataSource is a dependency not a parameter. There is no difference between dataSource and a repository or a service

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/claimsbyhearingvenue/ClaimsByHearingVenueReportParams.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class ClaimsByHearingVenueReportParams extends ReportParams {
+    String hearingType;
+    String currentUserFullName;
+
+    public ClaimsByHearingVenueReportParams(String caseTypeId, String dateFrom, String dateTo,
+                                            String hearingDateType,  String userFullName) {
+        super(caseTypeId, dateFrom, dateTo);
+        hearingType = hearingDateType;
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/claimsbyhearingvenue/ClaimsByHearingVenueReportParams.java:16]: the convention in every other class is to write
this.hearingDateType = hearingDateType
this.userFullName = userFullName

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/claimsbyhearingvenue/ClaimsByHearingVenueReport.java (hunk 1 above)
Comment (btensay) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/claimsbyhearingvenue/ClaimsByHearingVenueReport.java:21]: fixed.

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/claimsbyhearingvenue/ClaimsByHearingVenueReportParams.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package uk.gov.hmcts.ethos.replacement.docmosis.reports.claimsbyhearingvenue;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import uk.gov.hmcts.ethos.replacement.docmosis.reports.ReportParams;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class ClaimsByHearingVenueReportParams extends ReportParams {
+    String hearingType;
```
[the remark this answers is not in the document]
Comment (jthmcts) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/claimsbyhearingvenue/ClaimsByHearingVenueReportParams.java:10]: What is the benefit of using this annotation over just explicitly writing out private final? Reading your code means you have to know what the lombok annotation does. lombok has its uses but I question the value of @value

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/claimsbyhearingvenue/ClaimsByHearingVenueReportParams.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    String userFullName;
+
+    public ClaimsByHearingVenueReportParams(String caseTypeId, String dateFrom, String dateTo,
+                                            String hearingDateType,  String userFullName) {
+        super(caseTypeId, dateFrom, dateTo);
+        this.hearingType = hearingDateType;
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/claimsbyhearingvenue/ClaimsByHearingVenueReportParams.java:16]: You changed userFullName but not this???

Comment (btensay) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/claimsbyhearingvenue/ClaimsByHearingVenueReportParams.java]: You are right. Fixed.

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/claimsbyhearingvenue/ClaimsByHearingVenueReportParams.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package uk.gov.hmcts.ethos.replacement.docmosis.reports.claimsbyhearingvenue;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import uk.gov.hmcts.ethos.replacement.docmosis.reports.ReportParams;
+
+@Value
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/claimsbyhearingvenue/ClaimsByHearingVenueReportParams.java:7]: Personally I don't like this. I don't see the value of this over explicitly putting private final on your variableas

**Review by jthmcts — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/claimsbyhearingvenue/ClaimsByHearingVenueReportParams.java (hunk 5 above)
Comment (btensay) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/claimsbyhearingvenue/ClaimsByHearingVenueReportParams.java:7]: Ok. I can make it simpler and explicit with @Getter and @Setter.

**Pull request merged**